robosuite/wrappers/ik_wrapper.py

```diff
@@ -105,11 +105,12 @@ def step(self, action):
         # keep trying to reach the target in a closed-loop
         for i in range(self.action_repeat):
             ret = self.env.step(low_action)
-            velocities = self.controller.get_control()
-            if self.env.mujoco_robot.name == "sawyer":
-                low_action = np.concatenate([velocities, action[7:]])
-            else:
-                low_action = np.concatenate([velocities, action[14:]])
+            if i + 1 < self.action_repeat:
+                velocities = self.controller.get_control()
+                if self.env.mujoco_robot.name == "sawyer":
+                    low_action = np.concatenate([velocities, action[7:]])
+                else:
```

Comment:
change this else condition to an elif condition to check for name "baxter". throw an exception if the name if unrecognized.